bryceeppler/day-trading
# Docker setup

# Changes

Updated README and created templates for the following microservices:

- matching engine
- order creation
- order execution
- market data
- user management
- react frontend

For all the backend services:

- each service is a simple express API in `index.js`
- each service is connected to the same mongodb instance

The frontend/react container I've left empty aside from a basic dockerfile that should build a react app and serve it via nginx. I wasn't sure which framework/build pack we would use.

![image](https://github.com/bryceeppler/day-trading/assets/65413229/e1da3d7a-ee65-41ef-aa31-840fce4e08a4)


# Testing

1. Run `docker-compose up`, and verify all the containers started properly in docker desktop or via the command line.
2. You can hit each microservice's sample endpoints from your local machine by referencing the table below and sending `[GET] http://localhost:PORT/` 

| Port  | Service  |
|-------|---------------|
| 8001  | ms_user |
| 8002  | ms_order_execution |
| 8003  | ms_order_creation |
| 8004  | ms_matching_engine |
| 8005  | ms_market_data |


If you want to test a service independently:
1. Navigate to `/ms_your_service`
2. Build the image `docker build -t ms_your_service .`
4. Run the image `docker run -p 8000:3000 ms_your_service`
5. Send a GET request to http://localhost:8000/ to verify your service is working and port forwarding

![image](https://github.com/bryceeppler/day-trading/assets/65413229/fbb14dba-e531-463f-918e-082218db294e)
